pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -301,7 +327,15 @@ contract WeightedPool is BaseWeightedPool, WeightedPoolProtocolFees {
         WeightedPoolProtocolFees._updatePostJoinExit(postJoinExitInvariant);
     }
 
-    function _beforeProtocolFeeCacheUpdate() internal override {
+    /**
+     * @dev This function will revert when called within a Vault context (i.e. in the middle of a join or an exit).
+     *
+     * This function depends on the invariant value, which may be calculated incorrectly in the middle of a join or
+     * an exit because the state of the pool could be out of sync with the state of the vault. This makes the function
+     * unsafe to call in such contexts, and hence it is protected.
```

Comment:
```suggestion
     * an exit, because the state of the pool could be out of sync with the state of the vault.
```
I think this is enough

Reply:
I don't agree with this suggestion, we should explain that we're adding the modifier and why. Otherwise we'd just say 'this is dangerous', instead of 'this would be dangerous, *but* we fixt it this way'.

Reply:
how about now? https://github.com/balancer-labs/balancer-v2-monorepo/commit/ab7702eae6f4b17e50cdb89f1e52923e9d6d674e